specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,259 @@
+# OP Stack Manager
+
+[`op-contracts/v1.3.0`]: https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.3.0
+[Optimism Monorepo releases]: https://github.com/ethereum-optimism/optimism/releases
+[contract releases]: https://github.com/ethereum-optimism/optimism/blob/main/packages/contracts-bedrock/VERSIONING.md
+[interop specification]: ../interop/overview.md
+[ethereum-optimism/specs#94]: https://github.com/ethereum-optimism/specs/issues/94
+[admin roles]: https://specs.optimism.io/protocol/configurability.html#admin-roles
+[`SystemConfig` contract]: https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/SystemConfig.sol
+[`L2OutputOracle` contract]: https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L2OutputOracle.sol
+[standard configuration]: ../protocol/configurability.md
+[superchain registry]: https://github.com/ethereum-optimism/superchain-registry
+[ethereum-lists/chains]: https://github.com/ethereum-lists/chains
+[Batch Inbox]: ../protocol/configurability.md#consensus-parameters
+
+The OP Stack Manager is a contract that deploys the L1 contracts for an OP Stack chain in a single
+transaction. It provides a minimal set of user-configurable parameters to ensure that the resulting
+chain meets the [standard configuration] requirements.
+
+The version deployed is always a governance-approved contract release. The set
+of governance approved [contract releases] can be found on the
+[Optimism Monorepo releases] page, and is the set of releases named
+`op-contracts/vX.Y.Z`.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Deployment](#deployment)
+- [Interface](#interface)
+  - [`deploy`](#deploy)
+  - [Getter Methods](#getter-methods)
+- [Implementation](#implementation)
+  - [Batch Inbox Address](#batch-inbox-address)
+  - [Contract Deployments](#contract-deployments)
+- [Security Considerations](#security-considerations)
+  - [Chain ID Source of Truth](#chain-id-source-of-truth)
+  - [Chain ID Frontrunning](#chain-id-frontrunning)
+  - [Chain ID Value](#chain-id-value)
+  - [Proxy Admin Owner](#proxy-admin-owner)
+  - [Upgradeability (ABI Changes)](#upgradeability-abi-changes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Deployment
+
+The OP Stack Manager is deployed at `0xTODO`. It can be deployed as follows:
+
```

Comment:
Are you planning on using nick's method for deployment? Seems like the deployment section isn't useful as part of a spec.

Reply:
If so, pre-deterministic deployment won't be universally feasible for EVM chains which don't support pre-EIP 155.  IMO the best here would just leave it dependent to the deployer key + nonce.

Reply:
Deploying our standard Proxy with a specified salt is sufficient to get us the same OP Stack Manager address on each chain. This is not a hard requirement—I just thought having the same proxy address would be nice to have, but no objections to removing this section if we prefer.

Reply:
Ah I see - its going to be proxied, if you add that detail then i am fine with including it, given you include the details you mentioned in this comment

Reply:
If we do want to have deterministic deployments, it would be most ideal to use a create3 style deployment so that we can safely modify the solidity version in the future without it breaking things, this seems like a lot of work so perhaps we land on "preference for but ok with breaking deterministic addys"

Reply:
The spec now mentions this is a proxied contract. I'd like to leave the deployment info as TODO for now, and either flesh it out or delete it later on. It's a nice to have and not a critical aspect of the spec, so want to avoid blocking this PR on it, since this is the "experimental" section of the specs anyway. @tynes let me know if that's ok with you and we can resolve this thread

Reply:
I am fine with this